config.go

```diff
@@ -59,8 +60,10 @@ func newBuilder() *Builder {
 // It panics under the following circumstances:
 //     * target is not a struct pointer
 //     * struct contains unsupported fields (pointers, maps, slice of structs, channels, arrays, funcs, interfaces, complex)
-func (c *Builder) To(target interface{}) {
-	c.populateStructRecursively(target, "")
+// It either returns nil or a *Error, aggregating all errors encountered by Builder
+func (b *Builder) To(target interface{}) error {
```

Comment:
rather than having an `Err()` method on `Builder`, I opted to go for `error` as the return value for `To`.

Two reasons:
* Any of the `From` methods could easily be changed to lazy initialization. Its an implementation detail. If they become lazy, the `Err` method becomes useless.
* Returning `err` from `To` means that linters will now catch if the `err` is being ignored. If it was an `Err` method, linters wouldn't catch this